CBot/src/CBot/CBotFileUtils.cpp

```diff
@@ -364,4 +383,90 @@ bool ReadStream(std::istream& istr, std::ostream &ostr)
     return true;
 }
 
+bool WriteVarListAsArray(std::ostream &ostr, CBotVar* pVar, CBotContext& context)
+{
+    std::size_t numVar{0};
+    for (auto var = pVar; var != nullptr; var = var->GetNext())
+    {
+        ++numVar;
+    }
+    if (!WriteSize_t(ostr, numVar)) return false;
+
+    for (auto var = pVar; var != nullptr; var = var->GetNext())
+    {
+        if (!var->Save0State(ostr)) return false; // common header
+        if (!var->Save1State(ostr, context)) return false; // saves the data
+    }
+
+    return true;
+}
+
+bool ReadVarListFromArray(std::istream &istr, CBotVar*& pVar, CBotContext& context)
+{
+    delete pVar;
+    pVar = nullptr;
+    std::size_t numVar{0};
+    CBotVar* pPrev = nullptr;
+    CBotVarUPtr outVar{nullptr};
+
+    if (!ReadSize_t(istr, numVar)) return false;
+    while (numVar != 0 && 0 < numVar--)
```

Comment:
```suggestion
    for(;numVar != 0;--numVar)
```

1. `numVar--` is a post-increment. The expression evaluates to `numVar != 0 && 0 < numVar`. This can be further simplified to `numVar != 0` because `numVar` is unsigned (size_t)
2. `numVar` is not used in the body